openapi_python_client/parser/properties/property.py

```diff
@@ -0,0 +1,81 @@
+from typing import ClassVar, Optional, Set
+
+import attr
+
+from ... import utils
+
+
+@attr.s(auto_attribs=True, frozen=True)
+class Property:
+    """
+    Describes a single property for a schema
+
+    Attributes:
+        template: Name of the template file (if any) to use for this property. Must be stored in
+            templates/property_templates and must contain two macros: construct and transform. Construct will be used to
+            build this property from JSON data (a response from an API). Transform will be used to convert this property
+            to JSON data (when sending a request to the API).
+
+    Raises:
+        ValidationError: Raised when the default value fails to be converted to the expected type
+    """
+
+    name: str
+    required: bool
+    nullable: bool
+    _type_string: ClassVar[str] = ""
+    default: Optional[str] = attr.ib()
+    python_name: str = attr.ib(init=False)
+
+    template: ClassVar[Optional[str]] = None
+
+    def __attrs_post_init__(self) -> None:
+        object.__setattr__(self, "python_name", utils.to_valid_python_identifier(utils.snake_case(self.name)))
+
```

Comment:
Could you do something like this so you don't have to manually validate stuff when making one of the convertable types?

```suggestion
if self._type_string in CONVERTABLE_TYPES:
    object.__setattr__(self, "default", convert(self._type_string, value))

```

Reply:
Yes but the attrs documentation explicitly said to avoid stuff like this at all costs because the behavior is sorta hidden. Explicit > implicit or whatever. Of course I did cop out and use it for `python_name` because of how much work it was going to be to do that manually >.>